mops/src/hpe/cpu.tpp

```diff
@@ -277,3 +270,247 @@ void mops::homogeneous_polynomial_evaluation_vjp(
 
     throw std::runtime_error("Only up to polynomial order 10 is supported at the moment. Please contact the developers for more");
 }
+
+
+template<typename scalar_t, uint8_t polynomial_order>
+void _homogeneous_polynomial_evaluation_vjp_vjp_templated_polynomial_order(
```

Comment:
this function is pretty big, and without any explanation. Could you add comments about what's happening (not line by line, but explaining the overall steps of the code?)